From roel.kluin@gmail.com  Mon Aug 17 12:39:00 2009
From: Roel Kluin <roel.kluin@gmail.com>
Date: Fri, 14 Aug 2009 14:02:34 +0200
Subject: tty: icom: bit and/or confusion?
To: Andrew Morton <akpm@linux-foundation.org>, gregkh@suse.de
Message-ID: <4A85525A.9030601@gmail.com>


Previously, if any bit other than CMD_SND_BREAK was set, CMD_SND_BREAK
was not unset.

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/icom.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/serial/icom.c
+++ b/drivers/serial/icom.c
@@ -617,7 +617,7 @@ static void shutdown(struct icom_port *i
 	 * disable break condition
 	 */
 	cmdReg = readb(&icom_port->dram->CmdReg);
-	if ((cmdReg | CMD_SND_BREAK) == CMD_SND_BREAK) {
+	if (cmdReg & CMD_SND_BREAK) {
 		writeb(cmdReg & ~CMD_SND_BREAK, &icom_port->dram->CmdReg);
 	}
 }
